polyfill/lib/yearmonth.mjs

```diff
@@ -87,32 +100,33 @@ export class YearMonth {
       nanoseconds,
       'days'
     );
-    const lastDay = ES.DaysInMonth(year, month);
-    ({ year, month } = ES.SubtractDate(year, month, lastDay, years, months, weeks, days, disambiguation));
-    ({ year, month } = ES.BalanceYearMonth(year, month));
-    ({ year, month } = ES.RegulateYearMonth(year, month, disambiguation));
+
+    const Date = GetIntrinsic('%Temporal.Date%');
+    const calendar = GetSlot(this, CALENDAR);
+    const year = calendar.year(this);
+    const month = calendar.month(this);
+    const lastDay = calendar.daysInMonth(this);
+    const lastOfCalendarMonth = calendar.dateFromFields({ year, month, day: lastDay }, {}, Date);
+    const subtractedDate = calendar.minus(lastOfCalendarMonth, { ...duration, days }, options, Date);
+
     const Construct = ES.SpeciesConstructor(this, YearMonth);
-    const result = new Construct(year, month);
+    const result = calendar.yearMonthFromFields(subtractedDate, options, Construct);
     if (!ES.IsTemporalYearMonth(result)) throw new TypeError('invalid result');
     return result;
   }
   difference(other, options) {
     if (!ES.IsTemporalYearMonth(this)) throw new TypeError('invalid receiver');
     if (!ES.IsTemporalYearMonth(other)) throw new TypeError('invalid YearMonth object');
+    const calendar = GetSlot(this, CALENDAR);
+    if (calendar.id !== GetSlot(other, CALENDAR).id) {
+      other = new Date(GetSlot(other, ISO_YEAR), GetSlot(other, ISO_MONTH), calendar, GetSlot(other, REF_ISO_DAY));
+    }
     const largestUnit = ES.ToLargestTemporalUnit(options, 'years', ['weeks', 'days', 'hours', 'minutes', 'seconds']);
     const [one, two] = [this, other].sort(YearMonth.compare);
-    let years = two.year - one.year;
-    let months = two.month - one.month;
-    if (months < 0) {
-      years -= 1;
-      months += 12;
-    }
-    if (largestUnit === 'months') {
-      months += 12 * years;
-      years = 0;
-    }
-    const Duration = GetIntrinsic('%Temporal.Duration%');
-    return new Duration(years, months);
+
+    const smaller = { year: one.year, month: one.month, day: 1 };
+    const larger = { year: two.year, month: two.month, day: 1 };
+    return calendar.difference(smaller, larger, { ...options, largestUnit });
```

Comment:
Is it okay that the arguments passed to the calendar are merely Date-like, not real Dates?